### DISCUSSION TITLE
In `rsbuild@0.7.0`, the define that defines `process.end.NODE_ENV` will not be replaced

### DISCUSSION THREAD

**Discussion by peterroe** [web-infra-dev/rsbuild#3683] [General]
## Minimum reproduction

```js
// index.js
console.log('process.env.NODE_ENV', process.env.NODE_ENV);
```
```ts
// rsbuild.config.ts
export default {
  source: {
    entry: 'index.js',
    define: {
      'process.env.NODE_ENV': JSON.stringify('development'),
    },
  },
};
```
```js
// dist/index.js
console.log("process.env.NODE_ENV","production");
```

## expected

```diff
// dist/index.js
- console.log("process.env.NODE_ENV","production");
+ console.log("process.env.NODE_ENV","development");
```

**Comment by chenjiahan** [comment]
Hi, you can upgrade to Rsbuild v1.0 to fix this

**Reply by peterroe** [reply]
Thanks for the reminder, but for some reason I'm not only able to upgrade `rsbuild` to the latest version, is there any other solution to this?

**Reply by chenjiahan** [reply]
Try [v0.7.9](https://github.com/web-infra-dev/rsbuild/releases/tag/v0.7.9), it may contain the fix.

**Reply by peterroe** [reply]
But unfortunately, upgrading to 0.7.9, I'm having the same problem as here：https://github.com/web-infra-dev/rspack/issues/7022
